Unable to install git on 10.8.5: errors generated (undeclared identifier)

ENVIRONMENT:
- Clean MacOS 10.8.5 installed
- X-Code 5.11 installed
- Latest Command Line Tools for this X-Code installed
- Tigerbrew installed
- 'brew doctor' applied
- all git dependencies installed with NO errors/warnings

STEPS:
1. Attempt to install git via 'brew install git';
2. Wait for all dependencies installed;
3. See the git 2.43.0 package downloaded;
4. See Patching is done successfully;
5. See 'gmake install' attempted.

EXPECTED:
Git installed successfully.

ACTUAL:
```
==> Installing git
==> Downloading https://mirrors.edge.kernel.org/pub/software/scm/git/git-2.43.0.tar.xz
##################################################################################################################################################################################################################################### 100.0%
==> Patching
patching file sha1dc/sha1.c
patching file config.mak.uname
patching file Makefile
patching file contrib/credential/osxkeychain/git-credential-osxkeychain.c
==> gmake install prefix=/usr/local/Cellar/git/2.43.0 sysconfdir=/usr/local/etc CC=clang CFLAGS=-std=gnu99 LDFLAGS=
Last 15 lines from /Users/nazalur/Library/Logs/Homebrew/git/01.gmake:
        if (flag & kFSEventStreamEventFlagItemIsLastHardlink)
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                   kFSEventStreamEventFlagItemIsSymlink
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/FSEvents.h:399:3: note: 'kFSEventStreamEventFlagItemIsSymlink' declared here
  kFSEventStreamEventFlagItemIsSymlink = 0x00040000
  ^
compat/fsmonitor/fsm-listen-darwin.c:105:13: error: use of undeclared identifier 'kFSEventStreamEventFlagOwnEvent'; did you mean 'kFSEventStreamEventFlagUnmount'?
        if (flag & kFSEventStreamEventFlagOwnEvent)
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                   kFSEventStreamEventFlagUnmount
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/FSEvents.h:387:3: note: 'kFSEventStreamEventFlagUnmount' declared here
  kFSEventStreamEventFlagUnmount = 0x00000080, /* These flags are only set if you specified the FileEvents*/
  ^
3 errors generated.
make: *** [compat/fsmonitor/fsm-listen-darwin.o] Error 1

READ THIS: https://github.com/mistydemeo/tigerbrew/wiki/troubleshooting

These open issues may also help:
git: "dyld: incompatible cpu-subtype" ruby(10.4/G3) https://github.com/mistydemeo/tigerbrew/issues/272
building ghc 7.10.2 (dependency for git-annex) fails on Mountain Lion https://github.com/mistydemeo/tigerbrew/issues/658
```

--- a/Library/Formula/git.rb
+++ b/Library/Formula/git.rb
@@ -234,7 +234,7 @@ __END__
  		endif
  	endif
  
-+	ifneq ($(shell expr "$(uname_R)" : '[156789]\.'),2)
++	ifeq ($(shell test "`expr "$(uname_R)" : '\([0-9][0-9]*\)\.'`" -gt 13 && echo 1), 1)
  	# The builtin FSMonitor on MacOS builds upon Simple-IPC.  Both require
  	# Unix domain sockets and PThreads.
  	ifndef NO_PTHREADS

--- a/Library/Formula/git.rb
+++ b/Library/Formula/git.rb
@@ -212,7 +212,7 @@ __END__
  		OLD_ICONV = UnfortunatelyYes
  		NO_APPLE_COMMON_CRYPTO = YesPlease
  	endif
-+	ifeq ($(shell test "`expr "$(uname_R)" : '\([0-9][0-9]*\)\.'`" -lt 11 && echo 1),1)
++	ifeq ($(shell test "`expr "$(uname_R)" : '\([0-9][0-9]*\)\.'`" -lt 12 && echo 1),1)
 +		NO_REGEX=YesPlease
 +	else
 +		USE_ENHANCED_BASIC_REGULAR_EXPRESSIONS = YesPlease

